peps/pep-0735.rst

```diff
@@ -0,0 +1,613 @@
+------------------------------------------
+
+Poetry and PDM already offer a feature which each calls "Dependency Groups",
+but using non-standard data belonging to the ``poetry`` and ``pdm`` tools.
+(PDM also uses extras for some Dependency Groups, and overlaps the notion
+heavily with extras.)
+
+This PEP is not guaranteed to be a perfectly substitutable solution for the
+same problem space for each tool. However, the ideas are extremely similar, and
+it should be possible for Poetry and PDM to support at least some
+PEP-735-standardized Dependency Group configurations using their own Dependency
+Group nomenclature.
+
+A level of interoperability with Poetry and PDM is a goal of this PEP, but
+certain features and behaviors defined here may not be supported by Poetry and
+PDM. Matching the existing Poetry and PDM *semantics* for Dependency Groups is
+a non-goal.
+
+Dependency Groups are not Hidden Extras
+---------------------------------------
+
+One could be forgiven for thinking that Dependency Groups are just extras which
+go unpublished.
+
+However, there are two major features which distinguish them from
+extras:
+
+* they support non-package projects
+
+* installation of a dependency group does not imply installation of a package's
+  dependencies (or the package itself)
+
+Object Specification Does Not Allow for "PEP 508 Decomposition"
+---------------------------------------------------------------
+
+Poetry and PDM both allow for their object formats for dependency data to be
+used to decompose a PEP 508 specification into parts. For example, a dependency
+on ``requests==2.21.0`` could be written under these tools in a form like
+``{name = "requests", version = "==2.21.0"}``.
+
+This spec does not allow for such a structure. There is only one way to write
+down a PEP 508 dependency, as a string. This has two positive effects:
+
+* There are fewer ways of writing identical data
+
+* It is not possible, by design, to combine a PEP 508 specifier with other
+  key-value pairs in a dependency specifier
+
+Specification
+=============
+
+This PEP defines a new section (table) in ``pyproject.toml`` files named
+``dependency-groups``. The ``dependency-groups`` table contains an arbitrary
+number of user-defined keys, each of which has, as its value, a list of
+requirements specifiers (defined below).  These keys must match the following
+regular expression: ``[a-z0-9][a-z0-9-]*[a-z0-9]``. Meaning that they must be
+all lower-case alphanumerics, with ``-`` allowed only in the middle, and at
+least two characters long. These requirements are chosen so that the
+normalization rules used for PyPI package names are unnecessary as the names
+are already normalized.
+
+Requirements specifiers will use a definition based on standardized
+`Dependency Specifiers <https://packaging.python.org/en/latest/specifications/dependency-specifiers/>`__
+introduced in :pep:`508`.
+This PEP also proposes an object format to define a dependency, called a "PEP
```

Comment:
I would recommend against using the PEP number in the permanent name of this feature.